src/Navigation/Navigation.Xamarin.Forms/Service/NavigationService.registration.cs

```diff
@@ -185,20 +193,21 @@ private static void RegisterInternal(INavigationRegistration registration)
         }
 
         private static bool TryGetRegistration(
-            Type viewModelInterfaceType,
+            Type viewModelType,
             Type viewType,
             out INavigationRegistration registrationInfo)
         {
-            if (!IsRegistered(viewModelInterfaceType, viewType))
+            if (!IsRegistered(viewModelType, viewType))
             {
                 registrationInfo = null;
                 return false;
             }
 
 
             registrationInfo = NavigationRegistrations.FirstOrDefault(registration =>
-                registration.ViewModelType == viewModelInterfaceType &&
-                registration.ViewType == viewType);
+                registration.ViewModelType == viewModelType &&
+                viewType.IsAssignableFrom(registration.ViewType) &&
+                registration.Condition?.Invoke() != false);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5575746)